### PR TITLE
Add/support content block

### PR DESCRIPTION
**Pull request by vasyas**
Migrated content embed block source code from https://github.com/Automattic/wp-calypso/tree/trunk/apps/happy-blocks/src/support-content . Block allows embedding:
- WordPress.com Guide (support) pages
- WordPress.com and other bbPress forum topics.

In addition to migration:
- Combined two blocks into one
- Add the ability to embed content blocks from other forums besides WordPress.com
- Visual fixes to address 118-gh/Automattic/lighthouse-forums

## Testing instructions

1. Build the plugin, sync to sandox.
2. Enable the `blocks-everywhere/support-content`
3. Create a new topic in the forum, in the editor press / and search for "Content embed" block.
4. Insert _public_ forum topic URL, for example, https://wordpress.com/forums/topic/request-a-quote-on-menu/, and verify its content has been embedded in the editor.
5. Save the topic, and verify embedded block was rendered in the topic view.
6. Verify embed is inserted by pasting the forum topic URL directly into the editor.
7. Repeat steps 3-5 for the Support page, for example https://wordpress.com/support/domains/change-a-domain/

### CODE REVIEW

**Review by msurdi-a8c — APPROVED** [review]
Works great

File: src/support-content-block/confirm-content.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+						<NavigableMenu role="menu">
+							<MenuItem
+								variant="tertiary"
+								className="be-support-content-confirm-content__item"
+								onClick={ ( e ) => {
+									e.preventDefault();
+									confirm();
+								} }
+							>
+								Create embed
```
Comment (johngodley) [inline src/support-content-block/confirm-content.tsx]: These seem to be untranslated.

File: webpack.config.js [hunk 2]
```diff
@@ -1,15 +1,28 @@
-const MiniCssExtractPlugin = require("mini-css-extract-plugin");
+const MiniCssExtractPlugin = require( 'mini-css-extract-plugin' );
 const defaultConfig = require( '@wordpress/scripts/config/webpack.config' );
 
+// update resource asset to have public path inside plugin
+// required for correctly serving assets in production
+defaultConfig.module.rules
+	.filter( ( r ) => r.type === 'asset/resource' )
+	.forEach( ( r ) => {
+		r.generator.publicPath = '/wp-content/plugins/blocks-everywhere/build/';
```
Comment (johngodley) [inline webpack.config.js:9]: Can you expand more about what this is doing? I'm not sure we can guarantee that this path will always be valid.

**Review by johngodley — COMMENTED** [review]
All works great! Left a couple of minor things.

Are there existing blocks with the old name in production? We may need to migrate them.

I see there are a lot of references to wordpress.com in the code. Just to confirm, this also supports embedding forum posts from wordpress.org? Does it support forum posts on any site (for example, someone installs this on their own forum)?

The same for the support page embeds - are these specific to wp.com?

Depending on what is supported we should mention it in the readme.txt, especially about enabling it in the allowed blocks.

**Comment by johngodley** [comment]
Also, does this need topics to be enabled in the REST API?

File: webpack.config.js (hunk 2 above)
Comment (vasyas) [reply webpack.config.js:9]: Without this line, Webpack uses automatic `publicPath` generation based on JS bundle URL for required assets. It's working ok when `concat_js` is disabled. But with `concat_js` URL gets screwed (it misses the plugin path). To fix it, I have set the absolute path for the assets.

Not sure if there are better ways to fix it.

**Comment by vasyas** [comment]
> Are there existing blocks with the old name in production? We may need to migrate them.

None yet.

> I see there are a lot of references to wordpress.com in the code. Just to confirm, this also supports embedding forum posts from wordpress.org? Does it support forum posts on any site (for example, someone installs this on their own forum)?

> The same for the support page embeds - are these specific to wp.com?

> Depending on what is supported we should mention it in the readme.txt, especially about enabling it in the allowed blocks.

> Also, does this need topics to be enabled in the REST API?

Added answers to this + usage description to readme.txt.

@johngodley Please check.

**Comment by johngodley** [comment]
Thanks for the updates! I suspect the `publicPath` might come back to bite us in the future (outside of wp.com anyway), but I don't have an alternative right now so let's go with this and fix things when they happen.

:shipit:

**Comment by johngodley** [comment]
Let's plan to include this as part of the 1.10.0 release along with #49

**Pull request merged**